src/app/routes/cpsAsset/getInitialData.js

```diff
@@ -1,45 +1,39 @@
+import pipe from 'ramda/src/pipe';
+import fetchPageData from '../fetchPageData';
 import applyTimestampRules from '#lib/utilities/preprocessor/rules/timestamp';
 import addIdsToBlocks from '#lib/utilities/preprocessor/rules/addIdsToBlocks';
-import radioPageAddIdsToBlocks from '#lib/utilities/preprocessor/rules/radioPage/addIdsToBlocks';
 import applyBlockPositioning from '#lib/utilities/preprocessor/rules/blockPositioning';
 import parseInternalLinks from '#lib/utilities/preprocessor/rules/cpsAssetPage/convertToOptimoBlocks/blocks/internalLinks';
 import convertToOptimoBlocks from '#lib/utilities/preprocessor/rules/cpsAssetPage/convertToOptimoBlocks';
 import addHeadlineBlock from '#lib/utilities/preprocessor/rules/cpsAssetPage/addHeadlineBlock';
 import timestampToMilliseconds from '#lib/utilities/preprocessor/rules/cpsAssetPage/timestampToMilliseconds';
-import filterUnknownContentTypes from '#lib/utilities/preprocessor/rules/filterContentType';
-import filterEmptyGroupItems from '#lib/utilities/preprocessor/rules/filterEmptyGroupItems';
-import applySquashTopstories from '#lib/utilities/preprocessor/rules/topstories';
-import addIdsToItems from '#lib/utilities/preprocessor/rules/addIdsToItems';
-import filterGroupsWithoutStraplines from '#lib/utilities/preprocessor/rules/filterGroupsWithoutStraplines';
 import addSummaryBlock from '#lib/utilities/preprocessor/rules/cpsAssetPage/addSummaryBlock';
 import cpsOnlyOnwardJourneys from '#lib/utilities/preprocessor/rules/cpsAssetPage/cpsOnlyOnwardJourneys';
 import addBylineBlock from '#lib/utilities/preprocessor/rules/cpsAssetPage/addBylineBlock';
 
-export const cpsAssetPreprocessorRules = [
-  parseInternalLinks,
-  timestampToMilliseconds,
-  convertToOptimoBlocks,
+const formatPageData = pipe(parseInternalLinks, timestampToMilliseconds);
+const processOptimoBlocks = pipe(
   addHeadlineBlock,
   addSummaryBlock,
   applyTimestampRules,
   addBylineBlock,
   addIdsToBlocks,
   applyBlockPositioning,
   cpsOnlyOnwardJourneys,
-];
+);
+const processJson = async json => {
+  const formattedPageData = formatPageData(json);
+  const optimoBlocks = await convertToOptimoBlocks(formattedPageData);
+  return processOptimoBlocks(optimoBlocks);
```

Comment:
Following the mob, I've kinda worked out what niggled me about this secition, and the async stuff isn't it! This has been inherited from the existing code:
- These rules process a mix of JSON and the embedded XML payload: https://github.com/bbc/simorgh/pull/5544/files#diff-aee0535b19cad91ff9f3b13df06ce207R14
- This converts the embdeeded XML blocks in into JSON https://github.com/bbc/simorgh/pull/5544/files#diff-aee0535b19cad91ff9f3b13df06ce207R26
- The remaining rules process the JSON: https://github.com/bbc/simorgh/pull/5544/files#diff-aee0535b19cad91ff9f3b13df06ce207R15-R22

Just couldn't follow that from the code I could see. In an ideal world, it'd be nicer if we could convert the xml payload into JSON then just apply all the rules on JSON rather than a mix of JSON rules and XML rules. I'm not entirely sure in the scope of this PR how that could be made clearer.